### PR TITLE
Fix crash in InlineTemporaryCodeRefactoringProvider

### LINKED ISSUE #47058: 'InlineTemporaryCodeRefactoringProvider' encountered an error
**Version Used**: Microsoft Visual Studio 16.7.2

**Steps to Reproduce**:

I was scrolling over ~~`Binder_Statements_Tests.vb`~~ `Parser.vb`, not sure what exactly happened. So I don't have clear repro steps. However, the stack trace should give an insight on what happened.

**Expected Behavior**:

No crash

**Actual Behavior**:

```
System.InvalidOperationException : Sequence contains no matching element
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source,Func`2 predicate)
   at async Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InlineTemporary.InlineTemporaryCodeRefactoringProvider.VB$StateMachine_4_GetReferencesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InlineTemporary.InlineTemporaryCodeRefactoringProvider.VB$StateMachine_3_ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #47058

### CODE REVIEW

File: src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb [hunk 1]
```diff
@@ -74,9 +74,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InlineTemporary
                 Dim solution = document.Project.Solution
                 Dim findReferencesResult = Await SymbolFinder.FindReferencesAsync(local, solution, cancellationToken).ConfigureAwait(False)
 
-                Dim locations = findReferencesResult.Single(Function(r) Equals(r.Definition, local)).Locations
-                If Not locations.Any(Function(loc) semanticModel.SyntaxTree.OverlapsHiddenPosition(loc.Location.SourceSpan, cancellationToken)) Then
-                    Return locations
+                Dim referencedSymbol = findReferencesResult.SingleOrDefault(Function(r) Equals(r.Definition, local))
+                If referencedSymbol Is Nothing Then
+                    Return SpecializedCollections.EmptyEnumerable(Of ReferenceLocation)()
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: this seems like a bug.  how can we not have an entry for the very item we're doing FindReferences on?

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: @CyrusNajmabadi FYI, the C# version have the exact same null check.
Probably checking the C# file history in GitHub and seeing when the check was added would provide more information?

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: The C# check was introduced in https://github.com/dotnet/roslyn/commit/31fc1118a2a25c8e365674d9f334c6f4291d9bce#diff-009129e38bd68fdf74e5a3b09866f329

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: @allisonchou Pinging you as you're the author of this commit.

https://github.com/dotnet/roslyn/commit/31fc1118a2a25c8e365674d9f334c6f4291d9bce#diff-009129e38bd68fdf74e5a3b09866f329

Were you able to know why/when the result from FindReferencesAsync is empty?

Comment (CyrusNajmabadi) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: No.  We haven't been able to determine this.  It shoudl not happen.  I put through a PR to try to find some potential causes, but we've gotten no hits off of it yet.

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: @CyrusNajmabadi I hit this again while debugging.

image

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: @CyrusNajmabadi I hit this again while debugging.

image

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: @CyrusNajmabadi Note that it reproduces **constantly** for me at the same line.

Comment (CyrusNajmabadi) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: Can you please file a repro then :-)

Comment (Youssef1313) [reply src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb:79]: Unfortunately, couldn't reproduce again ;-(
It was reproducing constantly within the same debugging session.

**Pull request closed without merging**